### PR TITLE
doc: add Zigbee tools

### PR DESCRIPTION
**Pull request by greg-fer**
Fixes for T/Z user guide and samples. Added sniffer tool.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: samples/zigbee/light_switch/README.rst [hunk 1]
```diff
@@ -41,20 +41,22 @@ LED 4:
...
 .. note::
     If the brightness level is decreased to the minimum, the effect of turning on the light bulb might not be noticeable.
 
 Button 3:
-    When pressed while resetting the board, enables the Sleepy End Device behavior.
+    Pressing this button while resetting the board enables the Sleepy End Device behavior.
 
 Button 4:
-    When pressed, toggles the Finding & Binding mode on the local endpoint on or off.
+    Toggles the Finding & Binding mode on the local endpoint on or off.
```
Comment (wbober) [inline samples/zigbee/light_switch/README.rst:59]: Finding & binding is not supported.

File: samples/zigbee/light_switch/README.rst [hunk 2]
```diff
@@ -41,20 +41,22 @@ LED 4:
...
-    When pressed for a longer period of time, decreases the brightness of the **LED 4** of the connected light bulb.
+
+    Pressing this button for a longer period of time decreases the brightness of the **LED 4** of the connected light bulb.
 
 .. note::
     If the brightness level is decreased to the minimum, the effect of turning on the light bulb might not be noticeable.
 
 Button 3:
-    When pressed while resetting the board, enables the Sleepy End Device behavior.
+    Pressing this button while resetting the board enables the Sleepy End Device behavior.
```
Comment (wbober) [inline samples/zigbee/light_switch/README.rst:56]: This is not yet in.

Comment (wbober) [inline samples/zigbee/light_switch/README.rst:56]: This is not yet in. It will go in a separate PR with SED support.

File: doc/nrf/ug_zigbee.rst [hunk 3]
```diff
@@ -27,8 +27,8 @@ It includes all mandatory features of the |zigbee_version| specification and pro
 The stack comes with the following features:
 
 * Complete implementation of the Zigbee core specification and Zigbee Pro feature set.
-* Support for all device roles: coordinator, router, and end device.
-* Zigbee Cluster Library (a specification for end devices).
+* Support for all device roles: Coordinator, Router, and End Device.
+* Zigbee Cluster Library (a specification for End Devices).
```
Comment (tomchy) [inline doc/nrf/ug_zigbee.rst:31]: ZCL is for all Zigbee devices, not just for End Devices.

File: doc/nrf/ug_zigbee.rst [hunk 4]
```diff
@@ -27,13 +27,15 @@ It includes all mandatory features of the |zigbee_version| specification and pro
 The stack comes with the following features:
 
 * Complete implementation of the Zigbee core specification and Zigbee Pro feature set.
-* Support for all device roles: coordinator, router, and end device.
-* Zigbee Cluster Library (a specification for end devices).
+* Support for all device roles: Coordinator, Router, and End Device.
+* Zigbee Cluster Library.
 * Base Device Behavior.
 * Devices, described in former Zigbee Home Automation and Light Link profiles.
 * Zigbee Green Power.
```
Comment (wbober) [inline doc/nrf/ug_zigbee.rst:34]: Zigbee Green Power Proxy Basic

It is important to describe this precisely as we support only the mandatory subset.

**Comment by tomchy** [comment]
I went through the whole document and found two things, that may be improved:
- `* :option: CONFIG_ZBOSS_ERROR_PRINT_TO_LOG - Allows ZBOSS to log its errors; enabled by default.`: This is not quite right. It enables a module that prints out error name as the result of an assertion through ZB_ERROR_CHECK macro. This is never done by ZBOSS, but by the application.
- I could not find a note that ZBOSS logs are in a binary (hex dump) format.
